packages/app-store/routing-forms/trpc/findTeamMembersMatchingAttributeLogic.handler.ts

```diff
@@ -53,37 +53,47 @@ export const findTeamMembersMatchingAttributeLogicHandler = async ({
   const {
     teamMembersMatchingAttributeLogic: matchingTeamMembersWithResult,
     timeTaken: teamMembersMatchingAttributeLogicTimeTaken,
+    troubleshooter,
   } = await findTeamMembersMatchingAttributeLogicOfRoute(
     {
       response,
       routeId,
       form: serializableForm,
       teamId: form.teamId,
+      isPreview: !!isPreview,
     },
     {
       enablePerf: _enablePerf,
+      // Reuse same flag for enabling troubleshooter. We would normall use them together
+      enableTroubleshooter: _enablePerf,
       concurrency: _concurrency,
     }
   );
 
   if (!matchingTeamMembersWithResult) {
-    return null;
+    return {
+      troubleshooter,
+      result: null,
+    };
   }
   const matchingTeamMembersIds = matchingTeamMembersWithResult.map((member) => member.userId);
   const matchingTeamMembers = await UserRepository.findByIds({ ids: matchingTeamMembersIds });
 
-  console.log('_enablePerf, _concurrency', _enablePerf, _concurrency)
+  console.log("_enablePerf, _concurrency", _enablePerf, _concurrency);
   if (_enablePerf) {
     const serverTimingHeader = getServerTimingHeader(teamMembersMatchingAttributeLogicTimeTaken);
     ctx.res?.setHeader("Server-Timing", serverTimingHeader);
-    console.log('Server-Timing', serverTimingHeader)
+    console.log("Server-Timing", serverTimingHeader);
   }
 
-  return matchingTeamMembers.map((user) => ({
-    id: user.id,
-    name: user.name,
-    email: user.email,
-  }));
+  return {
+    troubleshooter,
+    result: matchingTeamMembers.map((user) => ({
+      id: user.id,
+      name: user.name,
+      email: user.email,
+    })),
```

Comment:
Just send what we are sending earlier in `result` prop along with `troubleshooter` prop